validator/src/main/java/io/smallrye/config/validator/BeanValidationConfigValidator.java

```diff
@@ -225,11 +225,15 @@ default void validatePropertyValue(
         }
     }
 
-    default void validateMappingClass(final Object mappingObject, final List<Problem> problems) {
+    default void validateMappingClass(
+            final Object mappingObject,
+            final List<Problem> problems,
+            final String currentPath) {
         final Set<ConstraintViolation<Object>> violations = getValidator().validate(mappingObject);
         for (ConstraintViolation<Object> violation : violations) {
-            problems.add(violation.getPropertyPath().toString().isEmpty() ? new Problem(violation.getMessage())
-                    : new Problem(violation.getPropertyPath() + " " + violation.getMessage()));
+            problems.add(violation.getPropertyPath().toString().isEmpty() ?
```

Comment:
The path should be built with `dot` as a separator.

Reply:
Sure, I'll make the modifications. Just to be clear, in the example provided in the test, the error message will go from `"server.cors.origins[2] someClassLevelCrossValidation If host is server3, then port value must be 3"` to `"server.cors.origins[2].someClassLevelCrossValidation If host is server3, then port value must be 3"`.

violation.getPropertyPath() returns a path object that starts at the level of the class that contains the validation which means that there is generaly not much information available.

Reply:
Ok, I guess I was confused a bit about how it was being built. I think that the first part of the message should always map to configuration property name, so it is clear for the user what to change.

In this case, the `someClassLevelCrossValidation` is not part of the configuration name, so we shouldn't append it with a `dot`, but it can be included in the message.

Reply:
Sure, I'll revert this portion to a space.